File: week7/p5.c
int towerOfHanoi(int n, int from, int to, int aux) {
    if (n == 0) return 0;
    int moves = towerOfHanoi(n - 1, from, aux, to);
    moves += 1;
    moves += towerOfHanoi(n - 1, aux, to, from);
    return moves;
}

File: week7/p5.py
class Solution:
    def towerOfHanoi(self, n, fromm, to, aux):
        if n == 0:
            return 0
        # Move n-1 disks from 'fromm' to 'aux'
        moves = self.towerOfHanoi(n-1, fromm, aux, to)
        # Move nth disk from 'fromm' to 'to'
        moves += 1
        # Move n-1 disks from 'aux' to 'to'
        moves += self.towerOfHanoi(n-1, aux, to, fromm)
        return moves